### PR TITLE
Cloud-provider-azure: ignore azclient package when running e2e builds

### PR DESCRIPTION
**Pull request by MartinForReal**
Cloud-provider-azure: ignore azclient package when running e2e builds

### CODE REVIEW

File: config/prow/config.yaml [hunk 1]
```diff
@@ -540,8 +540,12 @@ branch-protection:
               protect: true
             release-1.26:
               protect: true
+            release-1.27:
+              protect: true
             kubetest2-aks:
               protect: true
+            gh-pages:
+              protect: false
```
Comment (cblecker) [inline config/prow/config.yaml:548]: This change is not outlined/described in the PR body.

Comment (MartinForReal) [reply config/prow/config.yaml:548]: @cblecker it is listed in description. Thanks!

**Pull request merged**